tests/simulator/__init__.py

```
@@ -308,3 +293,6 @@ def clean_up_diags_dirs(self):
                 if os.path.exists(diag_dir):
                     shutil.rmtree(diag_dir)
         cpp_lib().mpi_barrier()
+
+    def run(self, diag_dir):
```

Ruff (0.8.2)

Redefinition of unused `run` from line 274

(F811)